arangod/RocksDBEngine/RocksDBMetadata.cpp

```diff
@@ -284,6 +323,33 @@ Result RocksDBMetadata::serializeMeta(rocksdb::WriteBatch& batch,
     }
   }
 
+  // Step 4. store the revision tree
+  if (coll.syncByRevision() && rcoll->needToPersistRevisionTree(maxCommitSeq)) {
+    rocksdb::SequenceNumber seq =
+        rcoll->serializeRevisionTree(output, maxCommitSeq, maxWorkTime);
+    appliedSeq = std::min(appliedSeq, seq);
+
+    rocksutils::uint64ToPersistent(output, seq);
+
+    key.constructRevisionTreeValue(rcoll->objectId());
+    rocksdb::Slice value(output);
+
+    rocksdb::Status s = batch.Put(cf, key.string(), value);
+    if (!s.ok()) {
+      LOG_TOPIC("ff234", WARN, Logger::ENGINES)
+          << "writing revision tree failed";
+      return res.reset(rocksutils::convertStatus(s));
+    }
+  } else {
+    key.constructRevisionTreeValue(rcoll->objectId());
+    rocksdb::Status s = batch.Delete(cf, key.string());
```

Comment:
is it correct to delete the revision tree if `rcoll->needToPersistRevisionTree(maxCommitSeq)` is false

Reply:
Good catch.